### PR TITLE
Add Karma tests, travis config.

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Refactor to run as karma tests.

### CODE REVIEW

**Comment by dmitrizagidulin** [comment]
BREAKING: The import signature on this lib is changed in this PR.

Before:
```
const X25519KeyPair = require('x25519-key-pair');
```

Now:
```
const {X25519KeyPair} = require('x25519-key-pair');
```

**Comment by dlongley** [comment]
This is probably good but we can't merge it yet without disrupting some things. I'll review more closely and approve when we can merge.

**Comment by dmitrizagidulin** [comment]
@dlongley I agree (re disruption). Though I think the way to fix that is:

1. Merge `initial` into `master`, and publish `v0.1.0`.
2. Rebase THIS PR onto `master` (and optionally publish it as `v0.2.0`).

That way, the `initial` branch (on which stuff is depending) will be left alone.

**Comment by dlongley** [comment]
@dmitrizagidulin -- that would be fine, but there's a high likelihood that someone will delete the initial branch when you do that, even if we put protection on it; because the pattern is... once released, delete `initial`. We can probably merge this at the beginning of next week.

**Comment by dlongley** [comment]
@davidlehn -- I'd like us to be able to pull this in today. Could you make sure anything using the old branches will get updated?

**Comment by davidlehn** [comment]
@dlongley Ok, is there a list of dependents?

File: tests/unit/index.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ describe('X25519KeyPair', () => {
 
       const result = key.verifyFingerprint(fingerprint);
       expect(result.valid).to.be.true;
-      expect(result.error).to.not.exist;
+      // expect(result.error).to.not.exist;
```
Comment (davidlehn) [inline tests/unit/index.js]: Why is this commented out?  Seems it should be enabled or removed.

File: tests/package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "test-karma": "karma start"
+  },
+  "keywords": [
+    "test",
+    "x25519-key-pair"
+  ],
+  "author": {
+    "name": "Digital Bazaar, Inc.",
+    "email": "support@digitalbazaar.com",
+    "url": "http://digitalbazaar.com/"
```
Comment (davidlehn) [inline tests/package.json]:
```suggestion
    "url": "https://digitalbazaar.com/"
```

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+language: node_js
+node_js:
+  - "8"
+  - "10"
```
Comment (davidlehn) [inline .travis.yml]: Add 12

Comment (dmitrizagidulin) [reply .travis.yml]: Added

File: tests/unit/index.js (hunk 1 above)
Comment (dmitrizagidulin) [reply tests/unit/index.js]: Fixed.

File: tests/package.json (hunk 2 above)
Comment (dmitrizagidulin) [reply tests/package.json]: Fixed.

**Comment by dmitrizagidulin** [comment]
Ok, now just gotta fix the `TypeError: publicKeyBuffer.equals is not a function` error in the browser.

**Comment by davidlehn** [comment]
That's the line that had the conflict?  Perhaps that's why the other version was there?

**Comment by dmitrizagidulin** [comment]
@davidlehn No, the conflict line was right before that.

This error is our old friend - Buffers in node.js has the .equals() method, but whatever the return type of base58.decode() from forge is, doesn't have an .equals() method.

**Comment by davidlehn** [comment]
@dmitrizagidulin I'm suggesting the other version was there to avoid the buffer issue:
```diff
-    const valid = fingerprintBuffer.slice(0, 1).toString('hex') === '25' &&                
-      publicKeyBuffer.equals(fingerprintBuffer.slice(1));                                  
+    const valid = fingerprintBuffer[0] === 37 &&                                           
+      publicKeyBuffer.toString() === fingerprintBuffer.slice(1).toString();                

```

**Comment by dmitrizagidulin** [comment]
@davidlehn Doh, I see... I don't remember why I changed it in the first place! (I _thought_ it was breaking some test?)

Anyways, reverted, both Node and Karma tests now work. We should be ready to merge.

**Comment by mattcollier** [comment]
Tests are passing locally and on CI.  Merging.

**Pull request merged**